src/content/docs/mobile-monitoring/new-relic-mobile/get-started/graphql-mobile-requests.mdx

```diff
@@ -0,0 +1,133 @@
+---
+title: Monitor GraphQL requests for mobile apps
+tags:
+    - GraphQL
+    - Mobile monitoring
+    - Get started
+metaDescription: "Monitor GraphQL requests as mobile HTTP requests."
+freshnessValidatedDate: "2023-11-29"
+---
+
+import mobileMobileGraphqlRequests from 'images/mobile_screenshot-full_mobile-graphql-requests.webp'
+
+import mobileGroupbyDropdown from 'images/mobile_screenshot-crop_group-by-dropdown.webp'
+
+import mobileRequestDetails from 'images/mobile_screenshot-crop_request-details.webp'
+
+
+In the dynamic realm of mobile app development, GraphQL has emerged as a transformative force, enabling developers to build powerful and streamlined data fetching tools. However, with the increasing complexity of GraphQL queries and the intricacies of mobile app architecture, the need for comprehensive monitoring has become even more important. New Relic can help you monitor GraphQL requests so you can:
+
+* Identify and optimize queries that are causing delays or performance slowdowns.
+
+* Proactively catch errors before they impact users.
+
+* Ensure data loads quickly and smoothly from end-user requests.
+
+<img
+    title="HTTP requests"
+    alt="Screenshot of the HTTP requests UI"
+    src={mobileMobileGraphqlRequests}
+/>
+
+<figcaption>
+    **[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Mobile > (select an app) > HTTP requests**: View your GraphQL requests based on the custom name of their function.
+</figcaption>
+
+## Get started [#get-started]
+
+By default, New Relic monitors all your GraphQL requests, but we recommend you manually configuring the name of your GraphQL requests to parse out requests that perform different actions and view performance data for the different "groups" of requests. This is because unlike REST APIs, all GraphQL requests go to one `/graphql` endpoint, which makes it difficult to parse out requests that perform different actions. By giving custom names to your requests, you'll get a a clearer picture of how your GraphQL API is being used and help you to identify any potential performance issues of specific request functions.
+
+
+To custom name your GraphQL requests, you'll need to:
+
+1. Determine the specific network request that New Relic should capture.
+
+2. Use the `addHTTPHeadersTracking method` to set the `X-APOLLO-OPERATION-NAME` header for the identified request. The value must meet the following requirements:
+
+* Be a string with a length between 1 and 1024 characters
+* Consist exclusively of [ASCII characters](https://www.ascii-code.com/characters)
+
+See the language-specific examples below based on your mobile app platform, making sure to replace `operationName` with your custom name.
```

Comment:
@ndesai-newrelic What do customers need to know before using this request? Where are they adding/running this code? What if they want to create multiple custom names for multiple request groups?

Reply:
@ally-sassman they can add multiple header names for tracking and they also need to same headers in their http request .

Reply:
@ndesai-newrelic Can you provide examples of how to add multiple header names? I  imagine this will be a common use case.

Reply:
@ndesai-newrelic Thank you! Can you add this to the three collapser examples for Java, Obj-c, and swift in another commit?

Reply:
kotlin example:

```kotlin
        val headers = mutableListOf<String>()
        headers.add("operationName")
        headers.add("operationId")

        NewRelic.addHTTPHeadersTrackingFor(headers);
```

``` kotlin
val request = Request.Builder()
    .url("https://speed.hetzner.de/1GB.bin")
    .addHeader("operationName", "getDataFromServer")
    .addHeader("operationId", "123-46-789")
    .build()
``

Reply:
objective -c 

```c
[NewRelic addHTTPHeadersTrackingFor:@[@"operationName", @"operationId"]];
```

```c
[mutableRequest addValue:@"LaunchList" forHTTPHeaderField:@"operationName"];
[mutableRequest addValue:@"123-456-789" forHTTPHeaderField:@"operationId"];
```

Reply:
swift

```swift
NewRelic.addHTTPHeadersTracking(for:["operationName", "operationId"])
```

```swift
var request = URLRequest(url: URL(string:"https://www.myMobileApp.com/graphQL")!)
request.addValue("LaunchList", forHTTPHeaderField: "operationName")
request.addValue("123-456-789", forHTTPHeaderField: "operationId")
URLSession.shared.dataTask(with: request)

Reply:
@ally-sassman i added examples here.